### DISCUSSION TITLE
Country ISO code

### DISCUSSION THREAD

**Discussion by dlondero** [fattureincloud/api#144] [Q&A] [answered]
Ciao, ho ricevuto questo errore mentre provavo a creare un `Client` passando `Cechia` come valore di `country`

```json
{"error":{"message":"Invalid request.","validation_result":{"data.country":["The data.country field is not valid."]}}}
```

[`Cechia`](https://github.com/unicode-org/icu/blob/dcd19ae9bc0914f8be439a2de1a5fbc17e7b5447/icu4c/source/data/region/it.txt#L94)  l'ho ottenuto come localizzazione italiana dell'iso code `CZ` ([ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2)) tramite l'[Intl Component](https://symfony.com/doc/current/components/intl.html#country-names) di Symfony che si basa su [ICU](https://github.com/unicode-org/icu). A quanto pare da `GET /info/countries` Fatture in Cloud supporta solo `Repubblica Ceca` che in ICU è presente solo come [variante](https://github.com/unicode-org/icu/blob/dcd19ae9bc0914f8be439a2de1a5fbc17e7b5447/icu4c/source/data/region/it.txt#L313).

Passando `Repubblica Ceca` come valore di `country` alla chiamata l'errore non compare ovviamente.

Le domande ora sono:
- vi è mai capitata in precedenza questa situazione?
- pensate di aggiungere `Cechia` tra le country supportate?
- forse ci sono anche altri casi?

**Comment by emajo** [comment, marked as answer]
Ciao @dlondero per risolvere questo problema ti basta utilizzare il campo `country_iso` invece del campo `country`, es. `"country_iso": "CZ",`.

Facci sapere se questo risolve il problema,
Grazie,
Emanuele.

**Reply by dlondero** [reply]
Ciao @emajo, grazie provo. Non mi pare di vederlo nella [documentazione](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/entities/clients) altrimenti ci avrei provato.

**Reply by dlondero** [reply]
@emajo notavo che nel php-sdk non esiste `country_iso` per il model `Entity`. C'è solo `country` https://github.com/fattureincloud/fattureincloud-php-sdk/blob/master/lib/Model/Entity.php#L76

**Reply by valmoz** [reply]
@dlondero verrà rilasciato con la prossima versione delle SDK.

**Reply by antonino-t** [reply]
Buongiorno @valmoz qui siamo completamente bloccati nella migrazione. Necessitiamo con urgenza la possibilità di usare `country_iso` nella SDK. Sinceramente è inaspettato che richiedete il nome intero della nazione anziché la ISO. Anche perché come dovremmo mandare il dato? in quale lingua, ad esempio: `Germania`,` Germany` o `Deutschland`? se in Italiano dovremmo tradurre tutte le nazioni in Italiano e ce ne sono alcune che hanno traduzioni diverse, ad esempio le nazioni con più di una parola all'interno. Insomma tutto il mondo del codice gira intorno alle ISO proprio per evitare questo problerma.

**Reply by valmoz** [reply]
Ciao @gotrens stiamo lavorando attivamente al prossimo rilascio delle SDK per introdurre quel parametro.
Dovrebbe essere pronto a breve, vi chiedo ancora un po' di pazienza.

**Reply by emajo** [reply]
Ciao @gotrens abbiamo appena rilasciato una nuova versione dell'SDK che comprende questa modifica.

**Comment by mrsteam-dev** [comment]
Buongiorno,
mi ricollego a questa domanda.
Ho un problema utilizzando proprio country_iso che passando ES inserisce isole canarie al posto di spagna.

**Reply by emajo** [reply]
Ciao @mrsteam-dev, effettivamente sia spagna che isole canarie fanno riferimento al medesimo iso 'ES', un possibile workaround potrebbe essere quello di tenere salvate le informazioni della chiamata `GET /info/detailed_countries` per poi fare una ricerca tramite iso e tirar fuori il name della country più appropriata, e quindi passare `country` al posto di `country_iso` all'interno del cliente.
Non è sicuramente una soluzione elegante ma non credo ce ne siano molte altre.

**Reply by mrsteam-dev** [reply]
Ciao @emajo, effettivamente non è una soluzione molto elegante.
La domanda però è: per quale motivo separare le Canarie? Se non c'è una motivazione valida, è assurdo tenere sotto lo stesso ISO 2 entità, con il rischio di generare degli errori.
